### PR TITLE
[terminal] removed eager resize on attach, enhanced xterm types (fixes #1127)

### LINKED ISSUE #1127: Terminal content restored incorrectly when an editor is open
Restoring a running terminal populates the widget with recent terminal output / history, so that it looks the same as before the client restarting / reconnecting. However in some cases the content of the terminal is restored incorrectly: it seems like the previous content is restored as if the terminal was 1 character wide. i.e. the content is restored but one character per line. 

It seems that this happens only when an editor is open: 

peek 2018-01-25 08-45

note: this seems to affect only the restored content: any new output in the terminal is as expected.

### PR DESCRIPTION
**Pull request by svenefftinge**
Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

File: packages/terminal/src/browser/xterm.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module "xterm" {
```
Comment (akosyakov) [inline packages/terminal/src/browser/xterm.d.ts]: copyright

**Comment by hexa00** [comment]
Why is fit() no longer needed ? v2 to v3 change?

Otherwise nice if it's fixed.

**Comment by svenefftinge** [comment]
I don't know why it is not needed. I just see that it is wrong to call `fit` during `onAfterActivate` because at that point in time the dimensions might be bad, depending on whether `resize` was fired already. Removing it fixes it in all scenarios I could test (restore layout, initial layout).

**Comment by hexa00** [comment]
OK, note that this is also called by `onAfterShow` , you would need to test having 2 terminals open and switching between them.

If that works ok too then LGTM :)

**Comment by svenefftinge** [comment]
Yes, that works as well.

**Pull request merged**